### ISSUE TITLE
BigQuery dialect: Unable to lex characters: '&'

### ISSUE THREAD

**Issue by jdub** [sqlfluff/sqlfluff#1051]
## Expected Behaviour

The BigQuery dialect should be able to lex and parse bitwise operators, such as `&`.

## Observed Behaviour / Steps to Reproduce

bitwise.sql:
```sql
select features & (1 << 1) = (1 << 1) as is_internal from whatever
```

`sqlfluff parse --dialect bigquery bitwise.sql`:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    column_reference:
[L:  1, P:  8]      |                        identifier:                           'features'
[L:  1, P: 16]      |                    unparsable:                               !! Expected: 'Nothing...'
[L:  1, P: 16]      |                        whitespace:                           ' '
[L:  1, P: 17]      |                        unlexable:                            '&'
[L:  1, P: 18]      |                        whitespace:                           ' '
[L:  1, P: 19]      |                        start_bracket:                        '('
[L:  1, P: 20]      |                        raw:                                  '1'
[L:  1, P: 21]      |                        whitespace:                           ' '
[L:  1, P: 22]      |                        raw:                                  '<'
[L:  1, P: 23]      |                        raw:                                  '<'
[L:  1, P: 24]      |                        whitespace:                           ' '
[L:  1, P: 25]      |                        raw:                                  '1'
[L:  1, P: 26]      |                        end_bracket:                          ')'
[L:  1, P: 27]      |                        whitespace:                           ' '
[L:  1, P: 28]      |                        raw:                                  '='
[L:  1, P: 29]      |                        whitespace:                           ' '
[L:  1, P: 30]      |                        start_bracket:                        '('
[L:  1, P: 31]      |                        raw:                                  '1'
[L:  1, P: 32]      |                        whitespace:                           ' '
[L:  1, P: 33]      |                        raw:                                  '<'
[L:  1, P: 34]      |                        raw:                                  '<'
[L:  1, P: 35]      |                        whitespace:                           ' '
[L:  1, P: 36]      |                        raw:                                  '1'
[L:  1, P: 37]      |                        end_bracket:                          ')'
[L:  1, P: 38]      |                        whitespace:                           ' '
[L:  1, P: 39]      |                        raw:                                  'as'
[L:  1, P: 41]      |                        whitespace:                           ' '
[L:  1, P: 42]      |                        raw:                                  'is_internal'
[L:  1, P: 53]      |            whitespace:                                       ' '
[L:  1, P: 54]      |            [META] dedent:
[L:  1, P: 54]      |            from_clause:
[L:  1, P: 54]      |                keyword:                                      'from'
[L:  1, P: 58]      |                whitespace:                                   ' '
[L:  1, P: 59]      |                from_expression:
[L:  1, P: 59]      |                    [META] indent:
[L:  1, P: 59]      |                    from_expression_element:
[L:  1, P: 59]      |                        table_expression:
[L:  1, P: 59]      |                            table_reference:
[L:  1, P: 59]      |                                identifier:                   'whatever'
[L:  1, P: 67]      |                    [META] dedent:
[L:  1, P: 67]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:  17 |  LXR | Unable to lex characters: '&'
L:   1 | P:  16 |  PRS | Found unparsable section: ' & (1 << 1) = (1 << 1) as is_internal'
```

Note that the same error exists using the `ansi` dialect, but I'm not sure if it _should_ be supported for ANSI SQL.

## Version

```
$ sqlfluff --version
sqlfluff, version 0.5.6

$ python3 --version
Python 3.8.10
```

## Configuration

None.

**Comment by alanmcruickshank** [comment]
Thanks for the bug report. What's the significance of the leading `&`? I found the syntax for `A & B` but not `& A`. Docs:  https://cloud.google.com/bigquery/docs/reference/standard-sql/operators

The rest should be simple to implement.

**Comment by barrywhart** [comment]
Resolved by #1053

**Comment by jdub** [comment]
Thanks so much @barrywhart!

```
$ pip3 install --upgrade sqlfluff==0.6.0a1
Collecting sqlfluff==0.6.0a1
  Downloading sqlfluff-0.6.0a1-py3-none-any.whl (284 kB)
…
```

(deletes `-- noqa` lines…)

```
$ sqlfluff lint
=== [dbt templater] Compiling dbt project...
$
```

Hooray! 😄
